src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -381,23 +384,48 @@ compute shader, for several combinations of video format, video color spaces and
         usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.STORAGE_BINDING,
       });
 
+      // Use display size of VideoFrame and video size of HTMLVideoElement as frame size. These sizes are presenting size which
+      // apply transformation in video metadata if any.
+      const frameWidth =
+        sourceType === 'VideoFrame'
+          ? (source as VideoFrame).displayWidth
+          : (source as HTMLVideoElement).videoWidth;
+      const frameHeight =
+        sourceType === 'VideoFrame'
+          ? (source as VideoFrame).displayHeight
+          : (source as HTMLVideoElement).videoHeight;
+
+      // Pick pixels from centers of color chunks.
+      const coordTopLeft =
+        'vec2<i32>(' + Math.floor(frameWidth / 4) + ', ' + Math.floor(frameHeight / 4) + ')';
```

Comment:
nit: these computations could be done in the shader directly instead of via string building in JS.

Reply:
Thanks! Done.